tests/flags.cpp

```diff
@@ -0,0 +1,50 @@
+#include "openxr/openxr.hpp"
+#include <type_traits>
+
+#include <gtest/gtest.h>
```

Comment:
so much of this (all of it?) could be done with static_assert and no testing framework, I think, since it should all be constexpr...